File: atcoder.jp/contests/abc135/tasks/abc135_b/main.rb
# frozen_string_literal: true

# @return [Array(Integer)]
def input_int_array
  gets.split.map(&:to_i)
end

N = gets.to_i
P = input_int_array

cnt = 0
P.each_with_index do |p, i|
  cnt += 1 if i + 1 != p
end

puts [0, 2].include?(cnt) ? 'YES' : 'NO'

File: atcoder.jp/contests/abc135/tasks/abc135_b/main.go
package main

import (
	"fmt"
)

func YesNo(b bool) string {
	if b {
		return "YES"
	} else {
		return "NO"
	}
}

func InputInt() int {
	var n int
	fmt.Scanf("%d", &n)
	return n
}

func InputIntArray(n int) []int {
	result := make([]int, n)
	for i := range result {
		fmt.Scanf("%d", &result[i])
	}
	return result
}

func main() {
	N := InputInt()
	P := InputIntArray(N)

	cnt := 0
	for i, p := range P {
		if i+1 != p {
			cnt++
		}
	}

	ans := cnt == 0 || cnt == 2

	fmt.Println(YesNo(ans))
}
